tools/flang2/flang2exe/llutil.cpp

```diff
@@ -2194,6 +2194,86 @@ metadata_args_need_struct(void)
   return ll_feature_metadata_args_struct(&llvm_get_current_module()->ir);
 }
 
+/**
+ * This function returns true for the types supported
+ * in function make_param_op
+ */
+bool
+should_preserve_param(SPTR sptr)
+{
+  bool should_preserve = false;
+  DTYPE dtype = DTYPEG(sptr);
+
+  switch (DTY(dtype)) {
+    case TY_ARRAY:
+    case TY_STRUCT:
+    case TY_BLOG:
+    case TY_SLOG:
+    case TY_LOG:
+    case TY_BINT:
+    case TY_SINT:
+    case TY_INT:
+    case TY_REAL:
+    case TY_INT8:
+    case TY_LOG8:
+    case TY_DBLE:
+    case TY_QUAD:
+    case TY_CMPLX:
+    case TY_DCMPLX:
+    case TY_CHAR:
+      should_preserve = true;
```

Comment:
Why not `return true`?

Reply:
Yes that would look great. Incorporating this comment.